_OBJECTS/_Object_Methods/js/main.js

```diff
@@ -0,0 +1,100 @@
+// Object methods, this
+
+
+// We write our code using objects that represent entities, that's called OOP
+
+
+// Method shorthand
+
+let user = {
+    sayHi: function() {
+        alert('Hello')
+    }
+};
+
+
+// method shorthand looks better
+
+let user = {
+    sayHi() { //same as sayHi: function()
+        alert('Hello')
+    }
+};
+
+
+// this in methods
+
+/*
+It's common that an object needs to access the information stored in the object to do its job
+For instance, the code inside user.sayHi() may need the name of the user
+
+To access the object, a method can use the this keyword
+The value of this is the object(the one used to call the method)
+ */
+
+
+ let user = {
+     name: "John",
+     age: 30,
+
+     sayHi() {
+         alert(this.name);
+     }
+ };
+
+ user.sayHi(); //John
+
+
+//  Here during the execution of user.sayHi(), the value of this will be user
+
+
+/*
+"this" is not bound
+
+this is undefined in strict mode. If we try to access this.name, there will be an error
+
+In non strict mode the value of this in such case will be the global object(window in a browser)
+This is a historical behaviour that "use strict fixes"
+
+
+Please not that usually a call of a function that uses this without an object is not normal, but rather
+a programming mistake. If a function has this, then it is usually meant to be called in the context of an object
+*/
+
+
+ function sayHi() {
+     alert(this);
+ }
+
+ sayHi(); //undefined
```

Comment:
Tested this, not quite undefined. What is `this` in this context? Pun NOT intended :)

Reply:
In strict mode, it should return undefined. But in non-strict mode, it should return the global window object. Simply because if a function has this, then it is usually meant to be called in the context of an object.